### PR TITLE
WT-9802 Generate stats for python tests by default

### PR DESCRIPTION
**Pull request by Sean04**

### CODE REVIEW

**Review by ddanderson — APPROVED** [review]
The change looks good.  I wonder if it has a noticeable performance impact (there's a reason that stats aren't on by default, and often there are specified as statistics=fast).  I know @lukech is trying to keep PR times down.

File: test/suite/wttest.py [hunk 1]
```diff
@@ -514,8 +514,19 @@ def extensionsConfig(self):
     def setUpConnectionOpen(self, home):
         self.home = home
         config = self.conn_config
+        
         if hasattr(config, '__call__'):
             config = self.conn_config()
+
+        # Collect all statistics for Python tests by default unless configured otherwise.
+        if not("statistics" in config):
```
Comment (Siddhartha8899) [inline test/suite/wttest.py:522]: I think creating this into a function would be neater.

File: test/suite/wttest.py [hunk 2]
```diff
@@ -514,8 +514,19 @@ def extensionsConfig(self):
...
+            if config != "":
+                config += ","
+            config += 'statistics=(all)'
+
```
Comment (Siddhartha8899) [inline test/suite/wttest.py:526]: comment here? 
`# Check if statistics are enabled and statistics_log is not configured.`

File: test/suite/wttest.py [hunk 3]
```diff
@@ -514,8 +514,19 @@ def extensionsConfig(self):
     def setUpConnectionOpen(self, home):
         self.home = home
         config = self.conn_config
+        
         if hasattr(config, '__call__'):
             config = self.conn_config()
+
+        # Collect all statistics for Python tests by default unless configured otherwise.
+        if not("statistics" in config):
+            if config != "":
```
Comment (Siddhartha8899) [inline test/suite/wttest.py:523]: I don't think there is a need for this, if the config is "", then we can just have config = `,statistics=(all)` and that won't be an issue.

File: test/suite/wttest.py (hunk 1 above)
Comment (Sean04) [reply test/suite/wttest.py:522]: We're only adding 4 lines here, I don't think a function is necessary.

**Review by Siddhartha8899 — APPROVED** [review]
Thanks for this change.

**Comment by Sean04** [comment]
> The change looks good. I wonder if it has a noticeable performance impact (there's a reason that stats aren't on by default, and often there are specified as statistics=fast). I know @lukech is trying to keep PR times down.

@ddanderson I've linked a patch build in the ticket. I don't see any noticeable increases in run times. Are there any other side effects for testing if a developer writes a new test without being aware that statistics are being gathered?

**Comment by ddanderson** [comment]
> Are there any other side effects for testing if a developer writes a new test without being aware that statistics are being gathered?

@Sean04 I'm not aware of any other side effects.  if you didn't see any performance changes, I'm good with this.  I would have accepted a percent or so, and it's quite possible it's less than that.

**Comment by Sean04** [comment]
evergreen merge

**Pull request merged**